pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -147,20 +149,35 @@ func parseConfig(getconfigCtx *getconfigContext, config *zconfig.EdgeDevConfig,
 		parseSystemAdapterConfig(getconfigCtx, config, source, forceSystemAdaptersParse)
 
 		if source != fromBootstrap {
-			parseBaseOS(getconfigCtx, config)
+			activateNewBaseOSFlag := parseBaseOS(getconfigCtx, config)
 			parseNetworkInstanceConfig(getconfigCtx, config)
 			parseContentInfoConfig(getconfigCtx, config)
 			parseVolumeConfig(getconfigCtx, config)
 
-			// parseProfile must be called before processing of app instances from config
-			parseProfile(getconfigCtx, config)
-			parseAppInstanceConfig(getconfigCtx, config)
+			// If we are activating a new baseOS, we need to skip processing
+			if source == fromController && activateNewBaseOSFlag {
+				// We need to activate the new baseOS
+				// before we can process the app instances
+				// which depend on the new baseOS
+				log.Noticef("parseConfig: Ignoring config as a new baseOS image is being activated")
+				return skipConfigUpdate
+			}
+
+			// We have handled the volumes, so we can now process the app instances. But we need to check if
+			// we are in the middle of a baseOS upgrade, and if so, we need to skip processing the app instances.
+			if getconfigCtx.configProcessingSkipFlagUpdate {
+				log.Noticef("parseConfig: Ignoring rest of the config as update flag set")
```

Comment:
Sorry reading linearly so my comments become messy.
 You can make this be part of the check on line 158 because you want to return skipConfigUpdate in that case as well.

Reply:
Fixed